tests/integration/cmd_push_test.go

```diff
@@ -53,6 +53,94 @@ var _ = Describe("odo push command tests", func() {
 
 	})
 
+	Context("Check memory and cpu config before odo push", func() {
+		It("Should work when both minmemory and maxmemory is set..", func() {
+
+			helper.CopyExample(filepath.Join("source", "nodejs"), context)
+
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", cmpName, "--project", project, "--context", context, "--app", appName)
+
+			helper.CmdShouldPass("odo", "config", "set", "minMemory", "100Mi", "--context", context)
+			helper.CmdShouldPass("odo", "config", "set", "maxmemory", "200Mi", "--context", context)
+			helper.CmdShouldPass("odo", "push", "--context", context)
+		})
+
+		It("Should work when memory is set..", func() {
+
+			helper.CopyExample(filepath.Join("source", "nodejs"), context)
+
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", cmpName, "--project", project, "--context", context, "--app", appName)
+
+			helper.CmdShouldPass("odo", "config", "set", "Memory", "300Mi", "--context", context)
+			helper.CmdShouldPass("odo", "push", "--context", context)
+		})
+
+		It("Should fail if minMemory is set but maxmemory is not set..", func() {
+
+			helper.CopyExample(filepath.Join("source", "nodejs"), context)
+
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", cmpName, "--project", project, "--context", context, "--app", appName)
+
+			helper.CmdShouldPass("odo", "config", "set", "minmemory", "100Mi", "--context", context)
+			output := helper.CmdShouldFail("odo", "push", "--context", context)
+			Expect(output).To(ContainSubstring("`minmemory` should accompany `maxmemory` or use `odo config set memory` to use same value for both min and max or try not passing any of them"))
+		})
+
+		It("should fail if maxmemory is set but minmemory is not set..", func() {
```

Comment:
Can these validation be done through UTs instead of over crowding the integration test file ?

Reply:
Since it covers multiple packages. Not just config package. It cannot be covered in UTs.

Reply:
The memory and cpu is set in config package but the overall/e2e validation of config parameters occur in component push package